### DISCUSSION TITLE
Ignore files/folders in pull request changes

### DISCUSSION THREAD

**Discussion by manoadamro** [community/community#30686] [Pull Requests]
I'm loving the file tree view in pull requests, it's incredibly useful! I do have one suggestion for a possible improvement.

In our monorepo, we have a `/docs` folder that contains the generated files needed for our pages deployment.
When a pull request triggers particular actions, those actions result in extra commits being pushed to the triggering branch. This means that each pull request will contain at least 60+ changed files, almost all of which are auto-generated by an action.

It would be handy to have a sort of `.gitignore` equivalent in the `.github` folder (eg: `.diff-ignore` or something) to define files or folders (using the same syntax as `.gitignore`) that are omitted from the changed files tree, as well as the changed files count. Perhaps there could be a message or note somewhere on the UI to note that `x` files are omitted due to these rules to make that clear. 

Alternatively, instead of completely omitting them, they could be "switched off" with a checkbox, where toggling the checkbox would toggle visibility of the item in the files tree (and all children down the hierarchy if it is a folder). Perhaps the ignore file could be used to define files or folders that are "switched off" by default in this case.

This would allow trunk-based development heavily automated in monorepos without making pull requests bigger than what is actually intended for review.

**Comment by ghost** [comment]
In our monorepo, we have a /docs folder that contains the generated files needed for our pages deployment.
When a pull request triggers particular actions, those actions result in extra commits being pushed to the triggering branch. This means that each pull request will contain at least 60+ changed files, almost all of which are auto-generated by an action.

**Comment by Strernd** [comment]
Just not rendering the files with `linguist-generated` isn't doing it for us. Our file tree is oftentimes 70/75 files generated, and we don't have an option to exclude them.

**Comment by yuriy-yarosh** [comment]
We do consider [this](https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github) official recommendation **Customizing how changed files appear on GitHub** to be Broken.

We store our `.yarn/cache` for yarn4 in our repo, and not being able to hide vendored deps is inconvenient, to say the least.

**Reply by rube-de** [reply]
over a year later, this is still broken..

**Reply by tloriato** [reply]
guess what?  another year

**Comment by frbvianna** [comment]
In addition to what @manoadamro described, it would also be useful if those files/folders hidden from the PR file tree did not count toward the total diff line count.

For instance, in the Go ecosystem, it is not uncommon to have dependencies in a `vendor` folder. It may be that a new feature requires updating one of those dependencies, then all of the updated vendor files are shown in the PR file tree, possibly lagging the review navigation even when the file diffs are not rendered. 

Then the actual feature may be comprised of 100 lines of code, but the PR shows something like `+693,851 −10`, fogging the actual stats.

**Reply by dynajoe** [reply]
We're having a similar issue with go `vendor` directory. Many thousands of lines changed showing up and slow loading diffs in browser.

Shouldn't creating `.gitattributes` with these contents work?

```
vendor/** linguist-generated
```

**Comment by lewisl9029** [comment]
I'd love to see an option added for generated files in the files filter dropdown that shows/hides all generated files marked by the `linguist-generated` directive in .gitattributes: 

![Screenshot 2024-05-08 at 2 56 58 PM](https://github.com/community/community/assets/6934200/cbc2f7e6-09c6-475d-a4f1-4fc0fafa0bc2)

Ideally there should also be a way to default it to unchecked (if it's not going to be the default to begin with).

**Comment by wdcs-meetsoni** [comment]
i also have jenkinsfile , kubernetes manifest file for different environments in different branched I don't want to merge them 
looking for some solution or walkaround
